docker/sync_repo

```diff
@@ -1,13 +1,20 @@
 #!/usr/bin/env bash
-  
+
+set -e
+
 REPO=$1
+BRANCH=$2
 
 # Don't sync if there's a .no_sync file
 [ -e .no_sync  ] || {
     # Sync notebook repo
     WORKDIR=$(mktemp -d git-XXXXXX)
-    git clone --depth 1 "$REPO" "$WORKDIR"
+    # git clone with backoff to give network whitelist time to start
+    NEXT_WAIT_TIME=0
+    until $(git clone --depth 1 --branch "$BRANCH" "$REPO" "$WORKDIR") ||  [ $NEXT_WAIT_TIME -eq 4 ]; do
+      sleep $(( NEXT_WAIT_TIME++ ))
+    done
     rm -rf "${WORKDIR}"/{.git,.github,.gitignore,LICENSE,README.md,DEAfrica_notebooks_template.ipynb} || true
     rsync -avuo "${WORKDIR}/" ~/
```

Comment:
```suggestion
    rsync --verbose --recursive "${WORKDIR}/" ~/
```

I wonder if we should make this simpler.

-a is quite opinionated, and -o preserves owner. -u updates times.

If we just do recursive and leave the rest defaults it may be happier?

Reply:
Also, expanding shorthand flags on scripts is good practice so we can use less brains in reading it!

Reply:
o and u seem to fail anyway, I'll make this change but I'll need to do some further testing to make sure the perms are still ok